marlowe-dashboard-client/src/Contract/State.purs

```diff
@@ -244,25 +244,29 @@ handleAction { followerAppId } (SetNickname nickname) = do
   assign _nickname nickname
   insertIntoContractNicknames followerAppId nickname
 
-handleAction input@{ currentSlot, walletDetails } (ConfirmAction namedAction) = do
-  executionState <- use _executionState
-  mMarloweParams <- use _mMarloweParams
-  for_ mMarloweParams \marloweParams -> do
-    let
-      contractInput = toInput namedAction
-
-      txInput = mkTx currentSlot (executionState ^. _contract) (Unfoldable.fromMaybe contractInput)
-    ajaxApplyInputs <- applyTransactionInput walletDetails marloweParams txInput
-    case ajaxApplyInputs of
-      Left ajaxError -> do
-        void $ query _submitButtonSlot "action-confirm-button" $ tell $ SubmitResult (Milliseconds 600.0) (Left "Error")
-        addToast $ ajaxErrorToast "Failed to submit transaction." ajaxError
-      Right _ -> do
-        assign _pendingTransaction $ Just txInput
-        void $ query _submitButtonSlot "action-confirm-button" $ tell $ SubmitResult (Milliseconds 600.0) (Right "")
-        addToast $ successToast "Transaction submitted, awating confirmation."
-        { dataProvider } <- ask
-        when (dataProvider == LocalStorage) (callMainFrameAction $ MainFrame.DashboardAction $ Dashboard.UpdateFromStorage)
+handleAction input@{ currentSlot, walletDetails, lastMarloweAppEndpointCall } (ConfirmAction namedAction) = do
+  case lastMarloweAppEndpointCall of
+    Nothing -> do
+      executionState <- use _executionState
+      mMarloweParams <- use _mMarloweParams
+      for_ mMarloweParams \marloweParams -> do
+        let
+          contractInput = toInput namedAction
+
+          txInput = mkTx currentSlot (executionState ^. _contract) (Unfoldable.fromMaybe contractInput)
+        callMainFrameAction $ MainFrame.DashboardAction $ Dashboard.SetLastMarloweAppEndpointCall $ Just ApplyInputs
+        ajaxApplyInputs <- applyTransactionInput walletDetails marloweParams txInput
+        case ajaxApplyInputs of
+          Left ajaxError -> do
+            void $ query _submitButtonSlot "action-confirm-button" $ tell $ SubmitResult (Milliseconds 600.0) (Left "Error")
+            addToast $ ajaxErrorToast "Failed to submit transaction." ajaxError
+          Right _ -> do
+            assign _pendingTransaction $ Just txInput
+            void $ query _submitButtonSlot "action-confirm-button" $ tell $ SubmitResult (Milliseconds 600.0) (Right "")
+            addToast $ successToast "Transaction submitted, awating confirmation."
+            { dataProvider } <- ask
+            when (dataProvider == LocalStorage) (callMainFrameAction $ MainFrame.DashboardAction $ Dashboard.UpdateFromStorage)
+    Just _ -> addToast $ errorToast "Application busy. Please try again in a moment." $ Just "We are still waiting to hear back from the previous action you submitted. Once you see a notification about that action, you can try to update this contract again."
```

Comment:
Is there a way we could add the request ID to the message passed in by the PAB? I see this as a short-term fix at best, but ideally we would not throttle asynchronous requests like this!

Reply:
Yeap, that would be great. @nau, is it possible to add a request id and set it in the last error?

Reply:
Yes, I thought we had agreed that was indeed the best plan. But since the backend can't do that yet, I thought this would do fine for now. Because we need something asap for the demo.

Bear in mind you'd have to try really hard to do anything before the response from the previous action comes back, so I'm really not that worried. But in any case, this is intended as a temporary solution until we can attach IDs to the requests.